pkg/kube/client.go

```diff
@@ -111,6 +111,28 @@ func (c *Client) Create(namespace string, reader io.Reader, timeout int64, shoul
 	if buildErr != nil {
 		return buildErr
 	}
+	for _, info := range infos {
+		// if resource is created under a namespace we have to make sure the namespace matches
+		if info.Namespace != namespace && info.Namespace != "" {
+			return fmt.Errorf("resource's namespace %s doesn't match the current namespace %s", info.Namespace, namespace)
+		}
+		if name != "" {
+			raw := info.Object.(runtime.Unstructured).UnstructuredContent()
+			label, ok := GetValue(raw, "metadata", "labels")
+			if ok {
+				if v, ok := label.(map[string]interface{}); ok {
+					v["io.cattle.field/appId"] = name
```

Comment:
worth making "io.cattle.field/appId" a constant? you use it 4 times could be helpful

Reply:
Done.

Reply:
could you combine this logic some how with createResourceWithAppID? feels like similar logic